Sources/MapboxNavigation/RouteLineController.swift

```diff
@@ -133,7 +133,18 @@ extension NavigationMapView {
             }
             set {
                 navigationMapView.routeLineTracksTraversal = newValue
-                if newValue {
+                updateLocationConsumer()
+            }
+        }
+        
+        func updateLocationConsumer() {
+            switch navigationMapView.userLocationStyle {
```

Comment:
Apparently this logic also depends on `navigationMapView.userLocationStyle`, which may be updated by the user at any moment. Shouldn't we hook this `updateLocationConsumer` to updating `userLocationStyle` too?

Reply:
Yes, you're right, thanks for the suggestion, I'll update this part.